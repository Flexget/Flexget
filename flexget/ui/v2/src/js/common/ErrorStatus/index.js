import { connect } from 'react-redux';
import StatusBar from 'common/StatusBar';
import { clearStatus } from 'store/status/actions';

export function mapStateToProps({ status }) {
  return {
    open: !!status.error,
    message: status.error && status.error.message,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(clearStatus()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatusBar);
