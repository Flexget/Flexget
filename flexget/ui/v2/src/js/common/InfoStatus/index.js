import { connect } from 'react-redux';
import StatusBar from 'common/StatusBar';
import { clearStatus } from 'store/status/actions';

export function mapStateToProps({ status }) {
  return {
    open: !!status.info,
    message: status.info,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    clearStatus: () => dispatch(clearStatus()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(StatusBar);
