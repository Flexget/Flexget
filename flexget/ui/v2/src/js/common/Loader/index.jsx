import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';

const Loader = (props) => {
  if (props.isLoading) {
    // While our other component is loading...
    if (props.timedOut) {
      // In case we've timed out loading our other component.
      return <div>Loader timed out!</div>;
    } else if (props.pastDelay) {
      // Display a loading screen after a set delay.
      return <CircularProgress />;
    }
    // Don't flash "Loading..." when we don't need to.
    return null;
  } else if (props.error) {
    // If we aren't loading, maybe
    return <div>Error! Component failed to load</div>;
  }
  // This case shouldn't happen... but we'll return null anyways.
  return null;
};

Loader.propTypes = {
  isLoading: PropTypes.bool,
  timedOut: PropTypes.bool,
  pastDelay: PropTypes.bool,
  error: PropTypes.object, // eslint-disable-line react/require-default-props
};

Loader.defaultProps = {
  isLoading: false,
  timedOut: false,
  pastDelay: false,
};

export default Loader;
