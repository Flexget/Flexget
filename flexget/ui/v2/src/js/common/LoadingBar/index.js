import { connect } from 'react-redux';
import LoadingBar from './LoadingBar';

export function mapStateToProps({ status }, { prefix = '@flexget' } = {}) {
  return {
    loading: !!(Object.keys(status.loading).find(type => type.startsWith(prefix))),
  };
}

export default connect(mapStateToProps)(LoadingBar);
export { LoadingBar };
