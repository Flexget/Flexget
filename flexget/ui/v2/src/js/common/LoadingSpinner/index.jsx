import React from 'react';
import PropTypes from 'prop-types';
import { Spinner, Wrapper } from './styles';

const LoadingBar = ({ loading }) => {
  if (loading) {
    return (
      <Wrapper>
        <Spinner />
      </Wrapper>
    );
  }
  return null;
};

LoadingBar.propTypes = {
  loading: PropTypes.bool,
};

LoadingBar.defaultProps = {
  loading: false,
};

export default LoadingBar;
