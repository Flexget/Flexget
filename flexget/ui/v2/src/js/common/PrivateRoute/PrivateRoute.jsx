import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';

const PrivateRoute = ({ component: Component, loggedIn, ...rest }) => (
  <Route
    {...rest}
    render={props => (
      loggedIn ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: '/login',
            state: { from: props.location }, // eslint-disable-line react/prop-types
          }}
        />
      )
    )}
  />
);

PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool.isRequired,
};

export default PrivateRoute;
