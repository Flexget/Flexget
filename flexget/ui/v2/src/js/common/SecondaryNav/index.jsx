import React from 'react';
import PropTypes from 'prop-types';
import { MuiThemeProvider } from 'material-ui/styles';
import { darkTheme } from 'theme';
import {
  SecondaryAppBar,
  SecondaryToolbar,
} from './styles';

const SecondaryNav = ({ children }) => (
  <MuiThemeProvider theme={darkTheme}>
    <SecondaryAppBar>
      <SecondaryToolbar>
        {children}
      </SecondaryToolbar>
    </SecondaryAppBar>
  </MuiThemeProvider>
);

SecondaryNav.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SecondaryNav;
