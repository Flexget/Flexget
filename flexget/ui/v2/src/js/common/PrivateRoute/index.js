import { connect } from 'react-redux';
import PrivateRoute from './PrivateRoute';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export default connect(mapStateToProps)(PrivateRoute);
export { PrivateRoute };
