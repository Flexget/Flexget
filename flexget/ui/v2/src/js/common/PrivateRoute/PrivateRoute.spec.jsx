import React from 'react';
import { mapStateToProps, PrivateRoute } from 'common/PrivateRoute';
import renderer from 'react-test-renderer';
import { router } from 'utils/tests';

const Component = () => <div />;

describe('common/PrivateRoute', () => {
  describe('PrivateRoute', () => {
    it('renders correctly when logged in', () => {
      const tree = renderer.create(
        router(<PrivateRoute component={Component} loggedIn />)
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    xit('renders correctly when logged out', () => {
      const tree = renderer.create(
        router(<PrivateRoute component={Component} loggedIn={false} />)
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return logged in if logged in', () => {
      expect(mapStateToProps({ auth: { loggedIn: true } })).toMatchSnapshot();
    });

    it('should return not logged in if logged out', () => {
      expect(mapStateToProps({ auth: { } })).toMatchSnapshot();
    });
  });
});
