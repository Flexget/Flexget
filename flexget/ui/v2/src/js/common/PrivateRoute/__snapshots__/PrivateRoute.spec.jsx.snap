// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/PrivateRoute PrivateRoute renders correctly when logged in 1`] = `<div />`;

exports[`common/PrivateRoute PrivateRoute renders correctly when logged out 1`] = `null`;

exports[`common/PrivateRoute mapStateToProps should return logged in if logged in 1`] = `
Object {
  "loggedIn": true,
}
`;

exports[`common/PrivateRoute mapStateToProps should return not logged in if logged out 1`] = `
Object {
  "loggedIn": false,
}
`;
